### PR TITLE
recipes/ivy-avy: Add

### PR DESCRIPTION
**Pull request by abo-abo**
### Brief summary of what the package does

Avy integration for Ivy.

### Direct link to the package repository

https://github.com/abo-abo/swiper

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [ ] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

File: recipes/ivy-avy [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(ivy-hydra :repo "abo-abo/swiper"
```
Comment (conao3) [inline recipes/ivy-avy:1]: Recipe name is `ivy-avy`, but this package is defined as `ivy-hydra`...?

**Comment by abo-abo** [comment]
@conao3 Fixed the typo. Thanks.

**Comment by riscy** [comment]
Everything looks good -- one small note is that someone might want to simultaneously use this package and use the <kbd>C-'</kbd> keybinding in ivy-minibuffer-map for something else.  In these sorts of cases, this line:
```elisp
(define-key ivy-minibuffer-map (kbd "C-'") 'ivy-avy)
```
could overwrite their customizations.

Just in case, would something like the following be possible instead?
```elisp
(unless (lookup-key ivy-minibuffermap (kbd "C-'"))
    (define-key ivy-minibuffer-map (kbd "C-'") 'ivy-avy))
```

**Comment by abo-abo** [comment]
@riscy Thanks, I've made your recommeded change.

**Pull request merged**